vortex-array/src/array/primitive/compute/fill.rs

```diff
@@ -27,10 +28,8 @@ impl FillForwardFn<PrimitiveArray> for PrimitiveEncoding {
 
         if validity.all_invalid() {
             match_each_native_ptype!(array.ptype(), |$T| {
-                return Ok(PrimitiveArray::new(
-                    Buffer::<$T>::zeroed(array.len()),
-                    Validity::AllValid
-                ).into_array());
+                let fill_value = Scalar::from($T::default()).cast(array.dtype())?;
+                return Ok(ConstantArray::new(fill_value, array.len()).into_array())
```

Comment:
this is funky that fill_forward would discard the buffer you allocated, my but alas a valid result